src/nspawn/nspawn.c

```diff
@@ -3205,9 +3205,10 @@ static int inner_child(
                 char **os_release_pairs) {
 
         _cleanup_free_ char *home = NULL;
-        size_t n_env = 1;
+        size_t n_env = 2;
         char *envp[] = {
                 (char*) "PATH=" DEFAULT_PATH_COMPAT,
+                (char*) "LANG=C.UTF-8",
```

Comment:
maybe conditionalize this on !arg_boot? PID 1 comes with a built-in default fallback, and we should probably use that if we call that.

(also, maybe add a comment, explaining briefly, and that C.UTF-8 should be a reasonably safe bet these days)